sdk/core/azure-core/README.md

```diff
@@ -27,30 +27,30 @@ from azure.core.pipeline.policies import (
     ProxyPolicy
 )
 
-class FooServiceClient():
+class FooServiceClient:
 
     @staticmethod
-    def create_config(credential, scopes, **kwargs):
+    def _create_config(credential, scopes, **kwargs):
         # Here the SDK developer would define the default
         # config to interact with the service
         config = Configuration(**kwargs)
-        config.headers_policy = HeadersPolicy({"CustomHeader": "Value"}, **kwargs)
-        config.user_agent_policy = UserAgentPolicy("ServiceUserAgentValue", **kwargs)
-        config.authentication_policy = BearerTokenCredentialPolicy(credential, scopes, **kwargs)
-        config.retry_policy = RetryPolicy(**kwargs)
-        config.redirect_policy = RedirectPolicy(**kwargs)
-        config.logging_policy = NetworkTraceLoggingPolicy(**kwargs)
-        config.proxy_policy = ProxyPolicy(**kwargs)
-        config.transport = kwargs.get('transport', RequestsTransport)
-
-    def __init__(self, transport=None, configuration=None, **kwargs):
-        config = configuration or FooServiceClient.create_config(**kwargs)
-        transport = config.get_transport(**kwargs)
+        config.headers_policy = kwargs.get('headers_policy', HeadersPolicy({"CustomHeader": "Value"}, **kwargs))
+        config.user_agent_policy = kwargs.get('user_agent_policy', UserAgentPolicy("ServiceUserAgentValue", **kwargs))
+        config.authentication_policy = kwargs.get('authentication_policy', BearerTokenCredentialPolicy(credential, scopes, **kwargs))
+        config.retry_policy = kwargs.get('retry_policy', RetryPolicy(**kwargs))
+        config.redirect_policy = kwargs.get('redirect_policy', RedirectPolicy(**kwargs))
+        config.logging_policy = kwargs.get('logging_policy', NetworkTraceLoggingPolicy(**kwargs))
+        config.proxy_policy = kwargs.get('proxy_policy', ProxyPolicy(**kwargs))
```

Comment:
Just suggesting for future work: I do think it might be worth thinking about having the policies exhaust kwargs as they go, and then checking at the end that kwargs is empty. This would catch e.g. the situation where a user supplies a mistypes a kwarg, or of they pass both `logging_policy` (instance) and logging policy kwargs at the same time.

Reply:
That's a very relevant point, we were discussing this with @johanste this morning. For instance, is that true that a kwarg is only for one policy? If no, your suggestion might get trouble.
Also, if you provide both "retry_policy" and "retry_total_retries", what you do? I advocate for "ignore retry_total_retries", but discussion is fair.

Reply:
Right, I was definitely assuming that the sets of kwargs per policy are disjoint. If that's not true then this approach would not work. Personally I think supplying both "retry_policy" and "retry_total_retries" should raise an error, but it would be hard to detect that it has even happened without draining kwargs along the way.